lib/runtime/wasmer/imports.go

```diff
@@ -1727,19 +1728,76 @@ func ext_offchain_http_request_start_version_1(context unsafe.Pointer, methodSpa
 	uri := asMemorySlice(instanceContext, uriSpan)
 
 	result := scale.NewResult(int16(0), nil)
+	var resultSetErr error
 
 	runtimeCtx := instanceContext.Data().(*runtime.Context)
 	reqID, err := runtimeCtx.OffchainHTTPSet.StartRequest(string(httpMethod), string(uri))
 
 	if err != nil {
 		logger.Errorf("failed to start request: %s", err)
-		_ = result.Set(scale.Err, nil)
+		resultSetErr = result.Set(scale.Err, nil)
 	} else {
-		_ = result.Set(scale.OK, reqID)
+		resultSetErr = result.Set(scale.OK, reqID)
 	}
 
-	enc, _ := scale.Marshal(result)
-	ptr, _ := toWasmMemory(instanceContext, enc)
+	if resultSetErr != nil {
+		logger.Errorf("failed to set the result data: %s", err)
+		return C.int64_t(0)
+	}
+
+	enc, err := scale.Marshal(result)
+	if err != nil {
+		logger.Errorf("failed to scale marshal the result: %s", err)
+		return C.int64_t(0)
+	}
+
+	ptr, err := toWasmMemory(instanceContext, enc)
+	if err != nil {
+		logger.Errorf("failed to allocate result on memory: %s", err)
+		return C.int64_t(0)
+	}
+
+	return C.int64_t(ptr)
+}
+
+//export ext_offchain_http_request_add_header_version_1
+func ext_offchain_http_request_add_header_version_1(context unsafe.Pointer, reqID C.int32_t, nameSpan, valueSpan C.int64_t) C.int64_t {
+	logger.Debug("executing...")
+	instanceContext := wasm.IntoInstanceContext(context)
+
+	name := asMemorySlice(instanceContext, nameSpan)
+	value := asMemorySlice(instanceContext, valueSpan)
+
+	runtimeCtx := instanceContext.Data().(*runtime.Context)
+	offchainReq := runtimeCtx.OffchainHTTPSet.Get(int16(reqID))
+
+	result := scale.NewResult(nil, nil)
+	var resultSetErr error
+
+	err := offchainReq.AddHeader(string(name), string(value))
+	if err != nil {
+		logger.Errorf("failed to add request header: %s", err)
+		resultSetErr = result.Set(scale.Err, nil)
+	} else {
+		resultSetErr = result.Set(scale.OK, nil)
+	}
+
+	if resultSetErr != nil {
```

Comment:
If `resultSetErr` is not nil, that means error in setting result. Then why print `err` instead of `resultSetErr`? If this makes sense, you would want to change this in `ext_offchain_http_request_start_version_1` as well.

You could simplify this like below
```suggestion
	resultMode := scale.OK

	err := offchainReq.AddHeader(string(name), string(value))
	if err != nil {
		logger.Errorf("failed to add request header: %s", err)
		resultMode = scale.Err
	}

	err = result.Set(resultMode, nil)
	if err != nil {
```

Reply:
Nice suggestion! Thanks!!